drivers/timer/Kconfig.cortex_m_systick

```diff
@@ -60,3 +60,14 @@ config CORTEX_M_SYSTICK_IDLE_TIMER
 
 	  The chosen IDLE timer node has to support setting alarm from the
 	  counter API.
+
+config CORTEX_M_SYSTICK_EXT_REF_CLOCK
+	bool
+	depends on CORTEX_M_SYSTICK
+	help
+	  The default clock source for the Cortex-M systick is the processor
+	  clock. Enabling this option switches to an implementation defined
+	  external reference clock. If unsure, say N.
+
+	  See section "B3.3.3 SysTick Control and Status Register, SYST_CSR"
+	  of the Armv7-M Architecture Reference Manual for details.
```

Comment:
This sounds like it belongs in the devicetree binding instead?

Reply:
Good point, I'll have a look into this

Reply:
@henrikbrixandersen I had a go at moving this to device tree

I was reluctant to change the dtsi files to have multiple compats, but that might be tidier i.e:
```
systick: timer@e000e010 {
	compatible = "arm,armv7m-systick", "arm,cortex-m-systick";
	reg = <0xe000e010 0x10>;
};
```